src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.PatternLocalRewriter.cs

```diff
@@ -619,6 +619,62 @@ static bool usesOriginalInput(BoundDecisionDagNode node)
                 }
             }
 
+            protected BoundDecisionDag ShareIndexerTemps(BoundDecisionDag decisionDag)
```

Comment:
Not satisfied with this impl (too subtle). We may want an separate indexerMap for each alternative branch.